packages/gatsby/src/query/page-query-runner.js

```diff
@@ -137,57 +136,18 @@ const calcInitialDirtyQueryIds = state => {
   return _.union(extractedQueriesThatNeedRunning, nodeAndNoDepQueries)
 }
 
-const makeQueryJobs = pathnames => {
-  const staticQueries = pathnames.filter(p => p.slice(0, 4) === `sq--`)
-  const pageQueries = pathnames.filter(p => p.slice(0, 4) !== `sq--`)
-  const state = store.getState()
-  const queryJobs = []
-
-  staticQueries.forEach(id => {
-    const staticQueryComponent = store.getState().staticQueryComponents.get(id)
-    const queryJob: QueryJob = {
-      id: staticQueryComponent.hash,
-      hash: staticQueryComponent.hash,
-      jsonName: staticQueryComponent.jsonName,
-      query: staticQueryComponent.query,
-      componentPath: staticQueryComponent.componentPath,
-      context: { path: staticQueryComponent.jsonName },
-    }
-    queryJobs.push(queryJob)
-  })
-
-  const pages = state.pages
-  pageQueries.forEach(id => {
-    const page = pages.get(id)
-    if (page) {
-      queryJobs.push(
-        ({
-          id: page.path,
-          jsonName: page.jsonName,
-          query: store.getState().components.get(page.componentPath).query,
-          isPage: true,
-          componentPath: page.componentPath,
-          context: {
-            ...page,
-            ...page.context,
-          },
-        }: QueryJob)
-      )
-    }
-  })
-  return queryJobs
-}
-
-const runInitialQueries = async activity => {
-  const pathnamesToRun = calcInitialDirtyQueryIds(store.getState())
-  if (pathnamesToRun.length === 0) {
-    return
+/**
+ * groups queryIds by whether they are static or page queries.
+ */
+const groupQueryIds = queryIds => {
+  const grouped = _.groupBy(queryIds, p => p.slice(0, 4) === `sq--`)
```

Comment:
I know this is bikeshedding but this wasn't super easy to follow (might be me though). Wdyt of making it explicit?
```suggestion
  const grouped = _.groupBy(queryIds, p => p.slice(0, 4) === `sq--` ? 'static' : 'page')
  return {
    staticQueryIds: grouped['static'] || [],
    pageQueryIds: grouped['page'] || [],
  }
```

Reply:
This is a great suggestion! It's much better than relying on `true`/`false`